lib/daru/index.rb

```diff
@@ -96,6 +96,27 @@ def [](*key)
         v
       end
     end
+    
+    def pos *args
+      # Causes Segmentation fault in array_wrapper
+      if args.first.is_a? Range
+        return args.first.to_a
```

Comment:
I know what caused the error. This line. 
```
> NMatrix.new [2], [5, nil]
```
Its not a NMatrix problem I think. Anything except numeric will cause an error.